src/backend/catalog/pg_attribute_encoding.c

```diff
@@ -972,7 +973,7 @@ check_attribute_encoding_entry_exist(Oid relid, bool *attnum_entry_present)
 
 	Assert(OidIsValid(relid));
 
-	MemSet(attnum_entry_present, false, sizeof(attnum_entry_present));
+	MemSet(attnum_entry_present, false, sizeof(*attnum_entry_present) * MaxHeapAttributeNumber);
```

Comment:
1. What's the point of using `sizeof(*attnum_entry_present)`?
2. Why not to move MemSet to the `attnumsWithEntries`  declaration and write the comment that attnum_entry_present should be initialized before the function call?

Reply:
> 2\. Why not to move MemSet under the `attnumsWithEntries`  declaration and write the comment that attnum_entry_present should be initialized before the function call?

like
```c
bool attnumsWithEntries[MaxHeapAttributeNumber] = {0};
```
?

Reply:
>like bool attnumsWithEntries[MaxHeapAttributeNumber] = {0}; ?

Maybe. On the other hand MemSet looks more conventional, but anyway

Reply:
True, since the function is only used once, `= {0}` is probably a better solution